components/director/internal/systemfetcher/loader.go

```diff
@@ -341,3 +342,20 @@ func retrieveRegion(labels map[string]interface{}) (string, error) {
 	}
 	return regionValue, nil
 }
+
+func areAppTemplatesEqual(appTemplate *model.ApplicationTemplate, appTemplateInput model.ApplicationTemplateInput) bool {
+	if appTemplate == nil {
+		return false
+	}
+
+	isAppInputJSONEqual := reflect.DeepEqual(appTemplate.ApplicationInputJSON, appTemplateInput.ApplicationInputJSON)
+	isLabelEqual := reflect.DeepEqual(appTemplate.Labels, appTemplateInput.Labels)
+	isWebhookEqual := (appTemplate.Webhooks == nil && appTemplateInput.Webhooks == nil) || reflect.DeepEqual(appTemplate.Webhooks, appTemplateInput.Webhooks)
```

Comment:
reflect.DeepEqual(nil, nil) returns true, so i think that (appTemplate.Webhooks == nil && appTemplateInput.Webhooks == nil) is redundant

Reply:
Not really. I've tested it previously. 
<img width="1517" alt="image" src="https://user-images.githubusercontent.com/17293744/210964910-aec02792-de4a-44ad-9e58-f8ec6422c81c.png">

Reply:
Thats because you are comparing two objects of different types. []model.Webhook and []*model.WebhookInput